clang/include/clang/Basic/CodeGenOptions.def

```diff
@@ -462,6 +462,9 @@ ENUM_CODEGENOPT(ZeroCallUsedRegs, llvm::ZeroCallUsedRegs::ZeroCallUsedRegsKind,
 /// non-deleting destructors. (No effect on Microsoft ABI.)
 CODEGENOPT(CtorDtorReturnThis, 1, 0)
 
+/// Whether to validate if a produced mangled name can be demangled with LLVM demangler.
+CODEGENOPT(NoDemanglingFailures, 1, 0)
```

Comment:
This is probably inverted.

Reply:
The macro signature is `CODEGENOPT(Name, Bits, Default)`, so I assume you're referring to the `Default` as being inverted (the bit count is 1, since this is a flag).

IIUC `0` means that the flag is off (= don't apply the assertion), `1` means the flag is on (= apply the assertion and validate that mangled names can be demangled). We don't want to set the flag to on by default, so `0` seems right.

Could you explain why you believe this inverted? I'm probably missing something.

Reply:
I mean the name.  We shouldn't have a true/false flag where the 'true' isn't "turn this thing on".  So this should be `DemanglingFailures`.

Reply:
Thank you, I've changed it to `DemanglingFailures`, so that `true` means "turn the diagnostic on".

Reply:
This has turned into `DiagnosticsDemanglerFailures` now. Resolving this thread to avoid duplicate discussions.

Reply:
I think `Diagnose` vs `Diagnostics`, since the point is to enable `diagnose`-ing of the feature makes more sense.